diff --git a/dap/src/dap-clients.adb b/dap/src/dap-clients.adb
index 9466a67..ae1dacc 100644
--- a/dap/src/dap-clients.adb
+++ b/dap/src/dap-clients.adb
@@ -679,7 +679,7 @@ package body DAP.Clients is
       then
          return Self.Selected_Thread;
       else
-         return Self.Stopped_Threads.Element (Self.Stopped_Threads.First);
+         return Self.Stopped_Threads (Self.Stopped_Threads.First);
       end if;
    end Get_Current_Thread;
 
