diff --git a/tools/gnatcov/gnatcov.gpr b/tools/gnatcov/gnatcov.gpr.new
index 65d5ac4..687b114 100644
--- a/tools/gnatcov/gnatcov.gpr
+++ b/tools/gnatcov/gnatcov.gpr.new
@@ -261,13 +261,11 @@ project Gnatcov is
 
    package Linker is
       for Required_Switches use
-        ("-l:libopcodes.a",
-         "-l:libbfd.a",
-         "-l:libiberty.a",
-         "-l:libsframe.a",
-         "-l:libz.a",
-         "-static-libstdc++",
-         "-static-libgcc");
+        ("-opcodes",
+         "-lbfd",
+         "-liberty",
+         "-lsframe",
+         "-lz");
    end Linker;
 
    package Binder is
