diff --git a/source/gpr/lsp-gpr_files.adb b/source/gpr/lsp-gpr_files.adb
index 1e99e97..ea9b8ca 100644
--- a/source/gpr/lsp-gpr_files.adb
+++ b/source/gpr/lsp-gpr_files.adb
@@ -1475,8 +1475,8 @@ package body LSP.GPR_Files is
    begin
       if Name'Length > 0 then
          return GPR2.Project.Create
-           (GPR2.Filename_Type (Name),
-            File.Search_Paths);
+           (Name => GPR2.Filename_Type (Name),
+            Paths => File.Search_Paths);
       else
          return Path_Name.Undefined;
       end if;
