diff --git a/langkit/diagnostics.py b/langkit/diagnostics.py
index 16ea74a..dccffe6 100644
--- a/langkit/diagnostics.py
+++ b/langkit/diagnostics.py
@@ -337,8 +337,8 @@ def get_structured_context() -> list[Location]:
 
 def get_current_location() -> Location:
     ctx = get_structured_context()
-    assert ctx
-    return ctx[0]
+    #    assert ctx
+    return ctx[0] if ctx else Location.nowhere
 
 
 def get_parsable_location(location: Location | L.LktNode) -> str:
